tools/extra/qpafilter/qpafilter.py

```diff
@@ -107,80 +112,99 @@ def write_sensor(self, sensor):
         upper_warning_id = self.threshold_map['Upper Warning']
         upper_fatal_id = self.threshold_map['Upper Fatal']
 
-        self.outfile.write(struct.pack(BLOB_STRUCT_FORMAT,
+        self.payload.write(struct.pack(BLOB_STRUCT_FORMAT,
                                        sensor_id,
                                        upper_warning_id,
                                        warning))
-        self.outfile.write(struct.pack(BLOB_STRUCT_FORMAT,
+        self.payload.write(struct.pack(BLOB_STRUCT_FORMAT,
                                        sensor_id,
                                        upper_fatal_id,
                                        fatal))
 
 
-class blob_reader:
+class blob_reader_v0:
     """Given an input file pointer, a sensor map, and a threshold map,
        provide an iterator interface to reading the raw sensor data
        items sequentially."""
+
+    version = 0
+
     def __init__(self, blobfp, sensor_map, threshold_map):
-        self.blob = blobfp.read()
-        self.start_marker_index = self.blob.find(BLOB_START_MARKER)
-        self.end_marker_index = self.blob.find(BLOB_END_MARKER)
+        self.raw_hdr = blobfp.read(BLOB_HEADER_SIZE)
+        self.blob_hdr = struct.unpack('<IIIII', self.raw_hdr)
```

Comment:
If i understood this correctly.. after this line, the self.blob_hdr will contain 5 elements in the list?  blob_hdr[0] -->header magic number, blob_hdr[1] --> payload length, blob_hdr[2] -->blob version, blob_hdr[3]-->payload crc,  and blob_hdr[4]->header crc

Reply:
Correct, you've got it.